File: java/dtdgen-master/src/main/java/com/philhanna/dtdgen/dtd/DTDPCDATAElement.java
package com.philhanna.dtdgen.dtd;

import com.philhanna.dtdgen.ElementModel;

/**
 * A DTDElementModel that represents the an element with parsed
 * character content (<code>#PCDATA</code>)
 */
public class DTDPCDATAElement extends DTDElementModel {

   /**
    * Creates a new element declaration over the specified element model
    * @param elementModel the element model
    */
   public DTDPCDATAElement(ElementModel elementModel) {
      super(elementModel);
   }

   /**
    * Generates the &lt;!ELEMENT <code>&lt;name&gt;</code> ( #PCDATA ) &gt; string
    */
   @Override
   public String toString() {
      final String output = String
            .format("<!ELEMENT %s ( #PCDATA ) >", getElementName());
      return output;
   }
}

File: dtdgen/dtd/dtd_pcdata_element.py
from dtdgen import ElementModel
from dtdgen.dtd import DTDElementModel


class DTDPCDATAElement(DTDElementModel):
    """A DTDElementModel that represents an element
    with parsed character content (#PCDATA)"""
    def __init__(self, element_model: ElementModel):
        """Creates a new PCDATA element over the specified element model"""
        super().__init__(element_model)

    def __str__(self):
        """Generates the <!ELEMENT> name (#PCDATA)> string"""
        output: str = f"<!ELEMENT {self.element_name} ( #PCDATA ) >"
        return output
